tests/results/DiffTest.py

```diff
@@ -626,3 +626,21 @@ def test_delete(self):
         self.uut.delete = True
         self.assertEqual(self.uut.modified, [])
         self.uut.delete = False
+
+    def test_linebreaks(self):
+        diff = Diff(['1'])
+        lines = ['1', '2', '3']
+        appended = diff._add_linebreaks(lines)
+        skip_eof = diff._generate_linebreaks(lines)
+        empty = diff._add_linebreaks([])
+        empty_skip = diff._generate_linebreaks([])
+
+        for line in skip_eof[:-1]:
+            self.assertRegex(line, '\d\n')
+        self.assertRegex(skip_eof[-1], '\d(?!\n)')
+
+        for line in appended:
+            self.assertRegex(line, '\d\n')
+
+        self.assertEqual(empty, [])
+        self.assertEqual(empty_skip, [])
```

Comment:
you could maybe simplify this a bit by inlining the call made at definition of `empty`. Then those things are closer together which actually belong together ;) That makes it especially easier to follow (as you don't have to think about 4 test variables at the same time)
```python
self.assertEqual(Diff._add_linebreaks([]), [])
self.assertEqual(Diff._add_linebreaks([]), Diff._generate_linebreaks[])
```